wgengine/netlog: send on closed channel

Flake https://github.com/tailscale/tailscale/actions/runs/19381409398/job/55464983366?pr=17563
```
panic: send on closed channel

goroutine 3382 [running]:
tailscale.com/wgengine/netlog.(*Logger).flushRecordLocked(0xc0001a6180)
	/home/runner/work/tailscale/tailscale/src/wgengine/netlog/netlog.go:394 +0x265
tailscale.com/wgengine/netlog.(*Logger).initRecordLocked.func1()
	/home/runner/work/tailscale/tailscale/src/wgengine/netlog/netlog.go:380 +0x165
created by time.goFunc
	/home/runner/.cache/tsgo/5c01b77ad0d27a8bd4ef89ef7e713fd7043c5a91/src/time/sleep.go:215 +0x45
FAIL	tailscale.com/wgengine/netlog	15.349s
```

/cc @dsnet @scottjab

--- a/wgengine/netlog/netlog_test.go
+++ b/wgengine/netlog/netlog_test.go
@@ -182,6 +182,7 @@ func TestUpdateRace(t *testing.T) {
 	group.Wait()
 	logger.mu.Lock()
 	close(logger.recordsChan)
+	logger.recordsChan = nil
 	logger.mu.Unlock()
 }
 

